solana/solana-ibc/programs/solana-ibc/src/transfer/impls.rs

```diff
@@ -1,26 +1,81 @@
+use anchor_lang::prelude::{CpiContext, Pubkey};
 use anchor_lang::solana_program::msg;
+use anchor_spl::token::{spl_token, Burn, MintTo, Transfer};
 use ibc::applications::transfer::context::{
     TokenTransferExecutionContext, TokenTransferValidationContext,
 };
 use ibc::applications::transfer::error::TokenTransferError;
 use ibc::applications::transfer::PrefixedCoin;
 use ibc::core::ics24_host::identifier::{ChannelId, PortId};
 use ibc::Signer;
+use uint::FromDecStrErr;
 
 // use crate::module_holder::IbcStorage<'_,'_>;
-use crate::IbcStorage;
+use crate::{id, IbcStorage, MINT_ESCROW_SEED};
 
-impl TokenTransferExecutionContext for IbcStorage<'_, '_> {
+impl TokenTransferExecutionContext for IbcStorage<'_, '_, '_, '_> {
     fn send_coins_execute(
         &mut self,
-        _from: &Self::AccountId,
-        _to: &Self::AccountId,
-        _amt: &PrefixedCoin,
+        from: &Self::AccountId,
+        to: &Self::AccountId,
+        amt: &PrefixedCoin,
     ) -> Result<(), TokenTransferError> {
-        //let sender_id = from.to_string();
-        //let receiver_id = to.to_string();
-        //let base_denom = amt.denom.base_denom.to_string();
-        todo!()
+        msg!(
+            "Sending coins from account {} to account {}, trace path {}, base \
+             denom {}",
+            from,
+            to,
+            amt.denom.trace_path,
+            amt.denom.base_denom
+        );
+        let sender_id = from.to_string();
+        let receiver_id = to.to_string();
+        let base_denom = amt.denom.base_denom.to_string();
+        let amount = amt.amount;
+        // Since amount is u256 which is array of u64, so if the amount is above u64 max, it means that the amount value at index 0 is max.
```

Comment:
This doesn’t track. `[1, 1, 1, 1]` is above `u64::MAX` and none of the elements are `u64::MAX`. I don’t think I understand what this condition is doing.

Reply:
Maybe I misunderstood how u256 is coded. Suppose I have a number 224, how would it be coded as u256?

Reply:
[U256 uses little-endian](https://docs.rs/primitive-types/0.12.2/primitive_types/struct.U256.html) so 224 would be encoded as `[224, 0, 0, 0]`.

Reply:
You can do `u64::try_from(amount)`. However, rather than converting to u64 we probably just need to deal with U256 numbers.

Reply:
`u64::try_from(U256::from(amount))` then. I still don’t see why we need to convert though. Just use Amount or U256 everywhere. Tokens have tendency to use large denominations so u64 may very well be insufficient.

Reply:
But we need to convert anyhow since, the transfer method only expects u64

Reply:
I see. So it’s Solana’s limitation which uses u64 for tokens. Could you add a comment explaining that where you check the amount?

Reply:
Sure sure I will add that.